fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/BitmapValueIndexTest.java

```diff
@@ -242,18 +265,176 @@ public void uniquenessViolationNotChecked() {
         }
     }
 
+    @Test
```

Comment:
I think this could use a few more tests. Namely:

1. Something that uses `not`. If there's a way to get an XOR, then that would be nice too, but I don't think we have that in our API (nor do we simplify "(a &or; b) &and; &not; (a &and; b)" into "a &oplus; b", I don't think).
1. Something that has a union of elements with disjoint physical bitmaps in the index. Something like (using a pseudo-SQL syntax) `SELECT bitmap(rec_no) FROM MySimpleRecord WHERE num_value_2 = 3 OR num_value_2 = 4` where all of the records with `num_value_2 = 3` are in one position group and all of the records where `num_value_2 = 3` are in a different one.

Reply:
Added a test for `NOT`, which can be supported.

Added a test for non-overlapping indexes.